### ISSUE TITLE
update to 1.0 api

### ISSUE THREAD

**Issue by bradrydzewski** [drone/drone-js#6]

**Comment by rohit-smpx** [comment]
Is this open for PRs and can I try my hand at this?

I think best source for updated endpoints would be https://github.com/drone/drone-go? Or are they documented somewhere else too. https://readme.drone.io/ doesn't seem to have info about deployment/promote endpoints

**Comment by bradrydzewski** [comment]
Yes, that would be great.  Most of the endpoints are documented at https://docs.drone.io/api/. The `drone-go` repository is also a good resource, however, some endpoints in `drone-go` are deprecated and should not be added to this library.

**Comment by rohit-smpx** [comment]
So, I realized later that this is only for browser. But since only the `_request` and `_subscribe` functions are dependent on the browser environment would it be okay if I moved those out and we made 2 builds one for node and one for browser? It's easier to test the results in node too.

And if I move those out, is it okay if I use another module for the node bundle ([got](https://www.npmjs.com/package/got)) to make requests? Or only use inbuilt methods?

**Comment by bradrydzewski** [comment]
upon further reflection I think a community maintained node package, in a separate repository, is probably the best course of action. When a package is hosted under the Drone organization it implies a long term commitment and support, which we are not able to guarantee at this time. I would fully support a community maintained node package, and would be happy to link to the package in our official documentation.

**Comment by rohit-smpx** [comment]
Okay. I'll try to get the fork up to date in my free time for my personal projects then. 
I'll update here once it becomes usable. And if more people end up contributing to it then I can maybe move it to an org or something with multiple maintainers.
